projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/dag_validator.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from collections import namedtuple
+from typing import Dict
+from typing import List
+
+from vdk.internal.core.errors import ErrorMessage
+from vdk.internal.core.errors import UserCodeError
+
+log = logging.getLogger(__name__)
+Error = namedtuple("Error", ["TYPE", "PERMISSION", "REQUIREMENT", "CONFLICT"])
+ERROR = Error(
+    TYPE="type", PERMISSION="permission", REQUIREMENT="requirement", CONFLICT="conflict"
+)
+allowed_job_keys = ["job_name", "team_name", "fail_meta_job_on_error", "depends_on"]
+required_job_keys = ["job_name", "depends_on"]
+
+
+class DagValidator:
+    def validate(self, jobs: List[Dict]):
+        validated_jobs = list()
+        for job in jobs:
+            validated_jobs = self._validate_job(job, validated_jobs)
+
+    def _raise_error(self, job, error_type, countermeasures):
+        raise UserCodeError(
+            ErrorMessage(
+                "",
+                "Meta Job failed due to a data job validation failure.",
+                f"There is a {error_type} error with job {job}.",
+                "The DAG will not be built and the meta data job will fail.",
+                countermeasures,
+            )
+        )
+
+    def _validate_job(self, job: Dict, validated_jobs):
+        self._validate_allowed_and_required_keys(job)
+        self._validate_job_name(job, validated_jobs)
+        self._validate_dependencies(job, validated_jobs)
+        self._validate_team_name(job)
+        self._validate_fail_meta_job_on_error(job)
+        log.info(f"Successfully validated job: {job['job_name']}")
+        validated_jobs.append(job["job_name"])
+        return validated_jobs
+
+    def _validate_allowed_and_required_keys(self, job):
+        if any(key not in allowed_job_keys for key in job.keys()):
+            forbidden_keys = [key not in allowed_job_keys for key in job.keys()]
```

Comment:
You can streamline the code (and remove minor duplication)
```
     forbidden_keys = [key not in allowed_job_keys for key in job.keys()]
     if forbidden_keys: 
           self._raise_error
     missing_keys = [key not in job for key in required_job_keys]
     if missing_keys:  
        ...

Reply:
Done.